### PR TITLE
Add content interface for notifying Juju when microk8s is installed;

### PR DESCRIPTION
**Pull request by ycliuhw**
Add a slot using `content` interface to notify Juju to pre-fetch Juju related images when `microk8s` is installed;

### CODE REVIEW

**Comment by ktsakalozos** [comment]
Hi @ycliuhw thank you for the PR.

Can you please not change the lines that are not related to your PR. Thanks.

File: snapcraft.yaml [hunk 1]
```diff
@@ -390,3 +390,10 @@ parts:
         echo "Skipped"
       fi
 
+slots:
+  juju-info:
+    interface: content
+    content: juju
+    name: juju
+    source:
+      read: [$SNAP/.juju-info/juju]
```
[the remark this answers is not in the document]
Comment (ycliuhw) [reply snapcraft.yaml]: Hi @ktsakalozos 
Sorry, the space changes was done by yaml linter. Just reverted those changes.
This file here is just used for triggering a notification to Juju side when `microk8s` is installed.
It does not have to exist.

[the remark this answers is not in the document]
Comment (wallyworld) [reply snapcraft.yaml]: $SNAP above refers to the juju snap and the named slot is part of the machinery of the snapd content interface used to trigger an action on the juju side when both snaps are installed on the same machine.

[the remark this answers is not in the document]
Comment (ktsakalozos) [reply snapcraft.yaml]: Thank you for the explanation.

**Comment by ktsakalozos** [comment]
You most probably want to merge this to the master branch. v1.16 only builds the v1.16 track whereas the master branch is used to build the latest track and any future releases.

File: snapcraft.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (wallyworld) [reply snapcraft.yaml]: So it seems we are not allowed to use plugs/slots with confined snaps. We've reached out to the snapd guys to see how to best proceed.

**Comment by ycliuhw** [comment]
Hi @ktsakalozos 
Sorry, we are double-checking if this `plugs/slots` feature is compatible with confined snaps.
Please ignore this PR for now, we will let you know once we got confirmed.

Thanks.

**Comment by ycliuhw** [comment]
created a new PR https://github.com/ubuntu/microk8s/pull/795, so close this one.

**Pull request closed without merging**